plugins/optimization-detective/tests/test-class-od-link-collection.php

```diff
@@ -335,6 +335,57 @@ public function data_provider_to_test_add_link(): array {
 				'expected_count'  => 0,
 				'error'           => 'Maximum width must be greater than zero and greater than the minimum width.',
 			),
+			'international_domain_name'                  => array(
+				'links_args'      => array(
+					array(
+						array(
+							'rel'  => 'preload',
+							'href' => 'https://xn--fsq.com/תמונה.jpg',
+							'as'   => 'image',
+						),
+					),
+				),
+				'expected_html'   => '
+					<link data-od-added-tag rel="preload" href="https://xn--fsq.com/תמונה.jpg" as="image">
+				',
+				'expected_header' => 'Link: <https://xn--fsq.com/%D7%AA%D7%9E%D7%95%D7%A0%D7%94.jpg>; rel="preload"; as="image"',
+				'expected_count'  => 1,
+				'error'           => '',
+			),
+			'non_ascii_path'                             => array(
+				'links_args'      => array(
+					array(
+						array(
+							'rel'  => 'preload',
+							'href' => 'https://example.com/חנות/תמונה.jpg',
+							'as'   => 'image',
+						),
+					),
+				),
+				'expected_html'   => '
+					<link data-od-added-tag rel="preload" href="https://example.com/חנות/תמונה.jpg" as="image">
+				',
+				'expected_header' => 'Link: <https://example.com/%D7%97%D7%A0%D7%95%D7%AA/%D7%AA%D7%9E%D7%95%D7%A0%D7%94.jpg>; rel="preload"; as="image"',
+				'expected_count'  => 1,
+				'error'           => '',
+			),
+			'multisite_subdirectory_non_ascii'           => array(
+				'links_args'      => array(
+					array(
+						array(
+							'rel'  => 'preload',
+							'href' => 'https://example.com/חנות/wp-content/uploads/2025/01/example.jpg',
```

Comment:
Let's add some tests for other characters in the URL to make sure they are re-encoded as expected, for example:

```suggestion
							'href' => 'https://example.com/חנות/wp-content/uploads/2025/01/example.jpg?ver=1+2',
```

Reply:
Added